### DISCUSSION TITLE
Errors migrating from docker to native

### DISCUSSION THREAD

**Discussion by novaclark** [netbox-community/netbox-docker#946] [Getting Help] [answered]
After a successful test of netbox-docker v3.4.4 I want to move to a native install. Another post on this site reported success with backing up and restoring the database, per [this page](https://github.com/netbox-community/netbox-docker/wiki/Troubleshooting#database-operations), but I get an error with that command:

`~/projects/netbox-docker$ docker-compose exec -T postgres sh -c 'echo $POSTGRES_USER'`
`ERROR: The Compose file './docker-compose.yml' is invalid because:`
`services.netbox-housekeeping.depends_on contains an invalid type, it should be an array`
`services.netbox-worker.depends_on contains an invalid type, it should be an array`

and I haven't been able to figure out what's causing it or how to resolve it. I found a post on [reddit ](https://www.reddit.com/r/networking/comments/ms5p2e/netbox_import_database_on_new_install/)that recommends this command:

`./manage.py dumpdata -a -o netbox-dump.json`

but that led to a series of errors that I ultimately wasn't able to resolve:

`~/projects/netbox-docker$ docker exec -it netbox-docker-netbox-1 sh`
`$ ./manage.py dumpdata -a -o netbox-dump.json`
`🧬 loaded config '/etc/netbox/config/configuration.py'`
`🧬 loaded config '/etc/netbox/config/extra.py'`
`🧬 loaded config '/etc/netbox/config/logging.py'`
`🧬 loaded config '/etc/netbox/config/plugins.py'`
`CommandError: Unable to serialize database: relation "extras_script" does not exist`
`LINE 1: SELECT COUNT(*) AS "__count" FROM "extras_script"`
`                                          ^`
`$ ./manage.py dumpdata -a -e extras.script -o netbox-dump.json`
`🧬 loaded config '/etc/netbox/config/configuration.py'`
`🧬 loaded config '/etc/netbox/config/extra.py'`
`🧬 loaded config '/etc/netbox/config/logging.py'`
`🧬 loaded config '/etc/netbox/config/plugins.py'`
`CommandError: Unable to serialize database: relation "extras_report" does not exist`
`LINE 1: SELECT COUNT(*) AS "__count" FROM "extras_report"`
`                                          ^`
`$ ./manage.py dumpdata -a -e extras.script -e extras.report -o netbox-dump.json`
`🧬 loaded config '/etc/netbox/config/configuration.py'`
`🧬 loaded config '/etc/netbox/config/extra.py'`
`🧬 loaded config '/etc/netbox/config/logging.py'`
`🧬 loaded config '/etc/netbox/config/plugins.py'`
`CommandError: Unable to serialize database: relation "django_rq_queue" does not exist`
`LINE 1: SELECT COUNT(*) AS "__count" FROM "django_rq_queue"`
 `                                         ^`
`$ ./manage.py dumpdata -a -e extras.script -e extras.report -e django.rq.queue -o netbox-dump.json`
`🧬 loaded config '/etc/netbox/config/configuration.py'`
`🧬 loaded config '/etc/netbox/config/extra.py'`
`🧬 loaded config '/etc/netbox/config/logging.py'`
`🧬 loaded config '/etc/netbox/config/plugins.py'`
`Traceback (most recent call last):`
`  File "/opt/netbox/netbox/./manage.py", line 10, in <module>`
`    execute_from_command_line(sys.argv)`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line`
`    utility.execute()`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 440, in execute`
`    self.fetch_command(subcommand).run_from_argv(self.argv)`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/management/base.py", line 402, in run_from_argv`
`    self.execute(*args, **cmd_options)`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/management/base.py", line 448, in execute`
`    output = self.handle(*args, **options)`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/management/commands/dumpdata.py", line 120, in handle`
`    excluded_models, excluded_apps = parse_apps_and_model_labels(excludes)`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/core/management/utils.py", line 100, in parse_apps_and_model_labels`
`    model = installed_apps.get_model(label)`
`  File "/opt/netbox/venv/lib/python3.10/site-packages/django/apps/registry.py", line 206, in get_model`
`    app_label, model_name = app_label.split(".")`
`ValueError: too many values to unpack (expected 2)`

My docker-fu is weak, but I've been doing a lot of reading and searching and not having any success so far. I'd appreciate some assistance with this. If nothing else, I think the wiki troubleshooting page should be updated to reflect whatever the problem is with the database backup command. If the problem is me, please enlighten me!

**Comment by kkthxbye-code** [comment, marked as answer]
Your docker-compose version is too old. Either update it if possible, or check if you have the plugin version installed `docker compose ...` instead of `docker-compose ...`.

Alternatively you can just remove the `depends_on` sections from the docker-compose file if you are migrating away from the docker setup anyway.

**Reply by novaclark** [reply]
Thanks. I replaced `docker-compose` with `docker compose` and the dump completed without error.

**Reply by kennethacurtis** [reply]
I'm using `docker compose` and still getting this error.
```
# docker --version
Docker version 25.0.1, build 29cf629
```
